File: Kotlin/LL_Kotlin_Essential_Training/src/main/java/com/aalzatea/courses/linkedinlearning/challenges/java/conditionals/Conditionals.java
package com.aalzatea.courses.linkedinlearning.challenges.java.conditionals;

public class Conditionals {

    public static void main(String... args) {
        var cardPoints = 7_000;

        String cardLevel;
        if (cardPoints >= 0 && cardPoints < 1000) {
            cardLevel = "pearl";
        } else if (cardPoints >= 1000 && cardPoints < 5_000) {
            cardLevel = "silver";
        } else if (cardPoints >= 5_000 && cardPoints < 10_000) {
            cardLevel = "gold";
        } else {
            cardLevel = "platinum";
        }

        var plural = cardPoints > 1 || cardPoints == 0 ? "s" : "";
        System.out.printf("You have %d point%s and are at the %s level.", cardPoints, plural, cardLevel);
    }
}

File: Kotlin/LL_Kotlin_Essential_Training/src/main/kotlin/com/aalzatea/courses/linkedinlearning/challenges/kotlin/conditionals/Conditionals.kt
package com.aalzatea.courses.linkedinlearning.challenges.kotlin.conditionals

fun main() {
    val cardPoints = 7_000

    //Replace this if with a when
    /*val cardLevel: String = if (cardPoints >= 0 && cardPoints < 1000) {
        "pearl"
    } else if (cardPoints >= 1000 && cardPoints < 5_000) {
        "silver"
    } else if (cardPoints >= 5_000 && cardPoints < 10_000) {
        "gold"
    } else {
        "platinum"
    }*/
    val cardLevel: String = when(cardPoints) {
        in 0..999 -> "pearl"
        in 1000..4999 -> "silver"
        in 5000..10000 -> "gold"
        else -> "platinum"
    }

    val plural = if(cardPoints > 1 || cardPoints == 0) "s" else ""
    println("You have $cardPoints point$plural and are at the $cardLevel level.")
}